iroh-gateway/src/handlers.rs

```diff
@@ -46,27 +53,56 @@ use crate::{
     templates::{icon_class_name, ICONS_STYLESHEET, STYLESHEET},
 };
 
+enum RequestPreprocessingResult {
+    ResponseImmediately(GatewayResponse),
```

Comment:
Can we rename this to `RespondImmediately` is a little bit more readable.

Reply:
Done